### ISSUE TITLE
[BUG] useModalForm doesn't backfill correct generics

### ISSUE THREAD

**Issue by vinclou** [pankod/refine#2550]
**Describe the bug**
Summary: Type inference for useModalForm seems to be broken, given below
generic types you would expect the hook to be able to infer and return the correct object;
But the return object as any is returned instead.

Example:
```tsx
	const {
		modalProps: closeModalProps,
		formProps: closeFormProps,
		show: closeModalShow
	} = useModalForm<Account, HttpError, InternalAccountUpdates>({
		action: 'edit',
		metaData: {
			httpMethod: 'patch'
		},
		warnWhenUnsavedChanges: true
	})
```

**Received Types**

`modalProps: any`
`formProps:  any`
`show: any`

**Expected behavior**

`modalProps: ModalProps`
`formProps:  FormProps<InternalAccountUpdates>`
`show: (id?: BaseKey) => void`

**To Reproduce**

Get the latest "@pankod/refine-antd": "^3.46.2" and use useModalForm(),
confirm types are not inferred correctly 

**Deps**

"@pankod/refine-antd": "^3.46.2"
"@pankod/refine-core": "^3.69.6"
"@pankod/refine-kbar": "^0.8.0"
"@pankod/refine-react-router-v6": "^3.33.0"

** Current Dirty(Low Effort) Workaround **

Custom type:
```tsx
type TUseCloseModalReturn = TUseModalFormReturnType<Account, HttpError,InternalAccountUpdates>
	const {
		modalProps: closeModalProps,
		formProps: closeFormProps,
		show: closeModalShow
	}: TUseCloseModalReturn
// assign this to the returned object
// where `TUseCloseModalReturn ===   useCloseModalReturn<_, _, _>` is the imported type from refine lib 
// no changes to the type made inference on the type doesn't seem to be broken
```

**Comment by omeraplak** [comment]
Hey @vinclou 👋,
Thank you for reporting issue! We'll fix it today 🚀

**Comment by omeraplak** [comment]
Hey @vinclou ,
We've released `@pankod/refine-antd@3.48.2`. Could you approve this fix?
